cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py

```diff
@@ -93,3 +93,75 @@ def parse_run_info_file(self) -> BclConvertRunInfo:
         LOG.info(f"Parsing Run info XML {self.run_info_path}")
         parsed_metrics = BclConvertRunInfo(tree=ET.parse(self.run_info_path))
         return parsed_metrics
+
+    def get_sample_internal_ids(self) -> List[str]:
+        """Return a list of sample internal ids."""
+        sample_internal_ids: List[str] = []
+        for sample_demux_metric in self.demux_metrics:
+            if sample_demux_metric.sample_project not in ["indexcheck", "Undetermined"]:
+                sample_internal_ids.append(sample_demux_metric.sample_internal_id)
+        return sample_internal_ids
+
+    def get_lanes_for_sample_internal_id(self, sample_internal_id: str) -> List[int]:
```

Comment:
Is it necessary to aggregate data for a sample across all lanes? I have not done that

Reply:
It is not used for data aggregation just to get the number of lanes on which a sample has been run. This is then used to fill the sequencing statistics model.